Feature/sri 1846 fhir client token server support

christav opened the pull request:
Hook up Rincon token server as alternative auth mechanism.

Commit:
Initial token auth works

Created token server auth class
Plumbed through settings from client

Commit:
auto-import token server auth

Added import in the __init__.py to trigger
auth registration automatically.

imalchenko commented on fhirclient/client.py:
```diff
@@ -30,64 +30,66 @@ class FHIRClient(object):
         with a backend system through a client_assertion parameter
         """
-        
+
+        self.settings = settings
+
```
Would a comment make sense here, to let users know what is expected to be passed in?

christav replied:
I added a section to the readme with these details, please look at that and let me know if it's sufficient.

Commit:
Use timezone aware now

Convert inner code to use timezone-aware
datetime for expiration check instead of
the naive version to support the token
server expiration.

Commit:
Don't require app id with explicit auth

Don't throw without an app id if we're giving
explicit auth in the settings. The appid
will be managed by whatever underlying
auth system is used.

Commit:
Add token server usage to readme
